### PR TITLE
[NodeBundle] allow to set icon for tree

### PR DESCRIPTION
**Pull request by sandergo90**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | 

The possibility already exists to change the icon in the sidebar tree of your page. This was already available by yml configuration. I've added a new interface, TreeIconInterface that can be implemented and that will return the icon that should be used. 

I've also changed some code formatting while on it.

### CODE REVIEW

File: src/Kunstmaan/NodeBundle/Helper/PagesConfiguration.php [hunk 1]
```diff
@@ -3,89 +3,175 @@
...
+     * @return mixed
+     */
     public function getIcon($refName)
     {
-        return $this->getValue($refName, 'icon');
+        return $this->getValue(
+            $refName,
+            'icon',
+            function ($page) {
+                /** @noinspection PhpDeprecationInspection */
```
Comment (Numkil) [inline src/Kunstmaan/NodeBundle/Helper/PagesConfiguration.php]: Why is there a noinspection deprecation annotation on your new code?

**Comment by sandergo90** [comment]
@Numkil removed ;)

File: src/Kunstmaan/NodeBundle/Helper/Menu/PageMenuAdaptor.php [hunk 2]
```diff
@@ -183,20 +183,20 @@ private function getTreeNodes(
...
+                $parentNodes = $repo->getAllParents($currentNode);
+                $this->activeNodeIds = [];
                 foreach ($parentNodes as $parentNode) {
                     $this->activeNodeIds[] = $parentNode->getId();
                 }
             }
         }
 
-        return (is_null($this->activeNodeIds) ? array() : $this->activeNodeIds);
+        return ($this->activeNodeIds ?? []);
```
Comment (treeleaf) [inline src/Kunstmaan/NodeBundle/Helper/Menu/PageMenuAdaptor.php]: If this is reverted to PHP5 code, then we can use this PR in whatever version..

Comment (sandergo90) [reply src/Kunstmaan/NodeBundle/Helper/Menu/PageMenuAdaptor.php]: Fixed ;)

**Pull request merged**